File: exercise/week04/java/src/main/java/construct/AgedSignalUpdater.java
package construct;

public class AgedSignalUpdater extends NormalArtefactUpdater {
    
    @Override
    public void update(Artefact artefact) {
        increase(artefact, 1);
        artefact.timeToLive--;
        if (artefact.timeToLive < 0) {
            increase(artefact, 1);
        }
    }
}

File: exercise/week04/c#/Construct/AgedSignalUpdater.cs
namespace Construct;

public class AgedSignalUpdater : NormalArtefactUpdater, IArtefactUpdater
{
    void IArtefactUpdater.Update(Artefact artefact)
    {
        Increase(artefact, 1);
        artefact.TimeToLive--;
        if (artefact.TimeToLive < 0)
            Increase(artefact, 1);
    }
}